grails-app/services/com/unifina/service/SignalPathService.groovy

```diff
@@ -101,6 +103,15 @@ class SignalPathService {
 			sp = (SignalPath) serializationService.deserialize(canvas.serialization.bytes)
 		}
 
+		// require read access to all streams when starting
+		// can be problematic when collaborating on shared canvas; though even then it makes sense to force
+		//   explicit read rights sharing to streams on that canvas
+		for (Stream s in sp.getStreams()) {
```

Comment:
I get the feeling that this should be encapsulated inside the `SignalPath` or `SignalPathRunner` class.

Reply:
yea it's a bit special-casing stream modules over other modules; what about modules that also use streams but are not stream modules? Now there's only SendToStream, and it does the authentication check in the module. Maybe it can be moved there if there comes lots of other modules. It would then have to be a startlistener and do that check in there.